sdk/textanalytics/ai-text-analytics/review/ai-text-analytics.api.md

```diff
@@ -351,7 +533,7 @@ export interface TextDocumentStatistics {
 export type TokenSentimentValue = "positive" | "mixed" | "negative";
 
 // @public
-export type WarningCode = "LongWordsInDocument" | "DocumentTruncated";
+export type WarningCode = "LongWordsInDocument" | "DocumentTruncated" | string;
```

Comment:
Hey @joheredi, is this related to a recent change in the code generator? This is the swagger definition: https://github.com/Azure/azure-rest-api-specs/blob/28d37f8847253787a56d39814e7ccd5194aeeabe/specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.3/TextAnalytics.json#L815

Reply:
Yeah this is the way the generator is doing "extensible" enums. I don't think we have reach a formal agreement on how to generate these.

@xirzec your proposal is to instead of generating this:

```typescript
export type WarningCode = "LongWordsInDocument" | "DocumentTruncated" | string;
export interface TextAnalyticsWarning {
    code: WarningCode;
    message: string;
}
```

Do this?

```typescript
export type KnownWarningCode = "LongWordsInDocument" | "DocumentTruncated";
export interface TextAnalyticsWarning {
    code: string;
    message: string;
}
```

Reply:
The issue with this way of modeling extensible enums is that it is exactly the same as `type WarningCode = string;`. The information about known values is only contained in the source code text and is fundamentally erased from the type when it comes down to set theory. Even if we added a `KnownWarningCodes` enum or type, the problem then is how to associate that type with the parameter in a discoverable way. It's not going to show up when you invoke completion in an IDE, for example.